backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala

```diff
@@ -1438,4 +1438,37 @@ abstract class ScalarFunctionsValidateSuite extends FunctionsValidateSuite {
         checkGlutenOperatorMatch[FilterExecTransformer](df)
     }
   }
+
+  test("date_trunc") {
```

Comment:
@zml1206 I was running part of this test by using your velox changes and it was giving incorrect result for year
Output for this 2015-07-22 10:01:40.123456 was coming to be 2014-12-31 16:00:00.0 instead of 2015-01-01 00:00:00.0
It may be due to timezone conversion, could you suggest if I am missing something?

Reply:
withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC")

Reply:
I had tried setting this and it gives correct results but I was confused that for a real workload also do we need to set it? Is setting of this config not needed if we pass timezone also to this function? 
From velox side I see that in call method of this function, timeZone_ comes out to be a nullptr, but gluten passes spark.sql.session.timeZone to velox as query config so I am confused as to why it comes to be a nullptr? Could you please share light on it as I am new to the timestamp part.

Reply:
https://github.com/zml1206/velox/blob/5fb41e85b66eaac7063c92ebc0643d8dc48722aa/velox/functions/sparksql/DateTimeFunctions.h#L547

Reply:
Here it comes to be nullptr when I added a logline to prine timezone, that's what caused the confusion.

Reply:
<img width="723" alt="image" src="https://github.com/user-attachments/assets/1a4810aa-3861-4aa6-a878-4ce3682cc293" />

Reply:
Thanks, then it seems some issue in my environment due to which it comes out to be null.

Reply:
For me initialize function was never getting called when I was running test from gluten, after changing the definition of initialize to this it is working fine.
FOLLY_ALWAYS_INLINE void initialize(
      const std::vector<TypePtr>& /*inputTypes*/,
      const core::QueryConfig& config,
      const arg_type<Varchar>* /*unitString*/,
      const arg_type<Timestamp>* /*timestamp*/) {
    std::cout << "Initializing DateTruncFunction" << std::endl;
    timeZone_ = getTimeZoneFromConfig(config);
  }

Reply:
Oh, my mistake, thank you.